File: 第三方/选择图片/swift和OC图片选择/AlbumPicker-master/AlbumPicker/AlbumPicker/LSYAlbumPicker_OC/ViewController/LSYAssetPreview.m
//
//  LSYAssetPreview.m
//  AlbumPicker
//
//  Created by okwei on 15/7/31.
//  Copyright (c) 2015年 okwei. All rights reserved.
//

#import "LSYAssetPreview.h"
#import "LSYAssetPreviewView.h"
#import "LSYAssetPreviewItem.h"
#import "LSYAlbumModel.h"
@interface LSYAssetPreview ()<UIScrollViewDelegate,LSYAssetPreviewNavBarDelegate,LSYAssetPreviewToolBarDelegate,LSYAssetPreviewItemDelegate>
@property (nonatomic,strong)UIScrollView *previewScrollView;
@property (nonatomic,strong) LSYAssetPreviewNavBar *previewNavBar;
@property (nonatomic,strong) LSYAssetPreviewToolBar *previewToolBar;
@property (nonatomic,strong) NSMutableArray *selectedAssets;
@end

@implementation LSYAssetPreview

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [self setAutomaticallyAdjustsScrollViewInsets:NO];
    [self prefersStatusBarHidden];
    self.selectedAssets = [NSMutableArray arrayWithArray:self.assets];
    [self.navigationController setNavigationBarHidden:YES animated:NO];
    [self.view addSubview:self.previewScrollView];
    [self.view addSubview:self.previewToolBar];
    [self.view addSubview:self.previewNavBar];
    [self setAssets];
}
-(UIScrollView *)previewScrollView
{
    if (!_previewScrollView) {
        _previewScrollView = [[UIScrollView alloc] initWithFrame:CGRectMake(0, 0,ScreenSize.width, ScreenSize.height)];
        [_previewScrollView setBackgroundColor:[UIColor blackColor]];
        [_previewScrollView setPagingEnabled:YES];
        [_previewScrollView setDelegate:self];
        [_previewScrollView setShowsHorizontalScrollIndicator:NO];
        [_previewScrollView setShowsVerticalScrollIndicator:NO];

    }
    return _previewScrollView;
}
-(LSYAssetPreviewNavBar *)previewNavBar
{
    if (!_previewNavBar) {
        _previewNavBar = [[LSYAssetPreviewNavBar alloc] init];
        [_previewNavBar setBackgroundColor:[UIColor colorWithRed:19/255.0 green:19/255.0 blue:19/255.0 alpha:0.75]];
        _previewNavBar.delegate = self;
    }
    return _previewNavBar;
}
-(LSYAssetPreviewToolBar*)previewToolBar
{
    if (!_previewToolBar) {
        _previewToolBar = [[LSYAssetPreviewToolBar alloc] init];
        [_previewToolBar setBackgroundColor:[UIColor colorWithRed:19/255.0 green:19/255.0 blue:19/255.0 alpha:0.75]];
        _previewToolBar.delegate = self;
        [_previewToolBar setSendNumber:(int)self.AlbumCollection.indexPathsForSelectedItems.count];
    }
    return _previewToolBar;
}

-(BOOL)prefersStatusBarHidden
{
    return YES;
}
-(void)setAssets
{
    [self.previewScrollView setContentSize:CGSizeMake(ScreenSize.width*self.assets.count, ScreenSize.height)];
    for (int i = 0; i<_assets.count; i++) {
        LSYAssetPreviewItem *previewItem = [[LSYAssetPreviewItem alloc] initWithFrame:CGRectMake(ScreenSize.width*i, 0, ScreenSize.width, ScreenSize.height)];
        previewItem.delegate = self;
        LSYAlbumModel *model = _assets[i];
        previewItem.asset = model.asset;
        [self.previewScrollView addSubview:previewItem];
    }
}
#pragma mark -UIScrollViewDelegate
-(void)scrollViewDidEndDecelerating:(UIScrollView *)scrollView
{
    int assetNumber = scrollView.contentOffset.x/ViewSize(self.view).width;
    LSYAlbumModel *model = self.assets[assetNumber];
    self.previewNavBar.selectButton.selected = model.isSelect;
    
}
#pragma mark -LSYAssetPreviewItemDelegate
-(void)hiddenBarControl
{
    _previewNavBar.hidden = !_previewNavBar.hidden;
    _previewToolBar.hidden = !_previewToolBar.hidden;
}
#pragma mark -LSYAssetPreviewNavBarDelegate
-(void)backButtonClick:(UIButton *)backButton
{
    [self.navigationController popViewControllerAnimated:YES];
}
-(void)selectButtonClick:(UIButton *)selectButton
{
    if (self.previewScrollView.isDecelerating) {
        return;
    }
    int assetNumber = self.previewScrollView.contentOffset.x/ViewSize(self.view).width;
    LSYAlbumModel *model = self.assets[assetNumber];
    model.isSelect = !model.isSelect;
    self.previewNavBar.selectButton.selected = model.isSelect;
    if (model.isSelect) {
        [self.AlbumCollection selectItemAtIndexPath:model.indexPath animated:NO scrollPosition:UICollectionViewScrollPositionNone];
        [self.selectedAssets addObject:model];
        [self.AlbumCollection.delegate collectionView:self.AlbumCollection didSelectItemAtIndexPath:model.indexPath];
    }
    else
    {
        [self.AlbumCollection deselectItemAtIndexPath:model.indexPath animated:NO];
        [self.selectedAssets removeObject:model];
        [self.AlbumCollection.delegate collectionView:self.AlbumCollection didDeselectItemAtIndexPath:model.indexPath];
    }
    [self.previewToolBar setSendNumber:(int)self.AlbumCollection.indexPathsForSelectedItems.count];
}
#pragma mark -LSYAssetPreviewToolBarDelegate
-(void)sendButtonClick:(UIButton *)sendButton
{
    if (self.delegate && [self.delegate respondsToSelector:@selector(AssetPreviewDidFinishPick:)]) {
        NSMutableArray *assets = [NSMutableArray array];
        for (LSYAlbumModel *model in self.selectedAssets) {
            [assets addObject:model.asset];
        }
        [self.delegate AssetPreviewDidFinishPick:assets];
    }
}
-(void)viewWillDisappear:(BOOL)animated
{
    [self.navigationController setNavigationBarHidden:NO animated:YES];
}
-(void)viewDidLayoutSubviews
{
    [super viewDidLayoutSubviews];
    [_previewNavBar setFrame:CGRectMake(0, 0, ViewSize(self.view).width, 64)];
    [_previewToolBar setFrame:CGRectMake(0, ViewSize(self.view).height-44, ViewSize(self.view).width, 44)];
}
-(void)dealloc
{
    
}
/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: 第三方/选择图片/swift和OC图片选择/AlbumPicker-master/AlbumPicker_Swift/AlbumPicker_Swift/LSYAlbumPicker_Swift/ViewController/LSYAssetPreview.swift
//
//  LSYAssetPreview.swift
//  AlbumPicker
//
//  Created by okwei on 15/8/6.
//  Copyright (c) 2015年 okwei. All rights reserved.
//

import Foundation
import UIKit
protocol LSYAssetPreviewDelegate:class{
    func AssetPreviewDidFinishPick(assets:NSArray)->Void
}
@objc class LSYAssetPreview: UIViewController {
    var assets :NSMutableArray!
    var albumPickerCollection:UICollectionView!
    weak var delegate:LSYAssetPreviewDelegate!
    var selectedAssets:NSMutableArray!
    var previewScrollView:UIScrollView!{
        didSet{
            previewScrollView.backgroundColor = UIColor.blackColor()
            previewScrollView.pagingEnabled = true
            previewScrollView.delegate = self
            previewScrollView.showsHorizontalScrollIndicator = false
            previewScrollView.showsVerticalScrollIndicator = false
            self.view.addSubview(previewScrollView)
        }
    }
    var previewNavBar:LSYAssetPreviewNavBar!{
        didSet{
            previewNavBar.backgroundColor = UIColor(red: 19.0/255.0, green: 19.0/255.0, blue: 19.0/255.0, alpha: 0.75)
            previewNavBar.delegate = self
            self.view.addSubview(previewNavBar)
        }
    }
    var previewToolBar:LSYAssetPreviewToolBar!{
        didSet{
            previewToolBar.backgroundColor = UIColor(red: 19.0/255.0, green: 19.0/255.0, blue: 19.0/255.0, alpha: 0.75)
            previewToolBar.delegate = self
            previewToolBar.setSendNumber(self.albumPickerCollection.indexPathsForSelectedItems()!.count)
            self.view.addSubview(previewToolBar)
        }
    }
    override func prefersStatusBarHidden() -> Bool {
        return true
    }
    private func setup() {
        self.automaticallyAdjustsScrollViewInsets = false
        self.prefersStatusBarHidden()
        self.selectedAssets = NSMutableArray(array: self.assets)
        self.navigationController?.navigationBarHidden = true
        self.previewScrollView = UIScrollView(frame: CGRectMake(0, 0, LSYSwiftDefine.ViewSize(self.view).width,LSYSwiftDefine.ViewSize(self.view).height))
        self.previewNavBar = LSYAssetPreviewNavBar()
        self.previewToolBar = LSYAssetPreviewToolBar()
        self.setAssets()
        
        
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        self.setup()
    }
    private func setAssets(){
        self.previewScrollView.contentSize = CGSizeMake(LSYSwiftDefine.ViewSize(self.view).width * CGFloat(self.assets.count), LSYSwiftDefine.ViewSize(self.view).height)
        for i in 0 ..< self.assets.count {
            let previewItem :LSYAssetPreviewItem = LSYAssetPreviewItem(frame: CGRectMake(LSYSwiftDefine.ViewSize(self.view).width * CGFloat(i),0, LSYSwiftDefine.ViewSize(self.view).width, LSYSwiftDefine.ViewSize(self.view).height))
            previewItem.delegate = self
            let model:LSYAlbumModel = self.assets[i] as!LSYAlbumModel
            previewItem.asset = model.asset
            self.previewScrollView.addSubview(previewItem)
        }
    }
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        self.navigationController?.setNavigationBarHidden(false, animated: true)
    }
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        self.previewNavBar.frame = CGRectMake(0, 0, LSYSwiftDefine.ViewSize(self.view).width,64)
        self.previewToolBar.frame = CGRectMake(0, LSYSwiftDefine.ViewSize(self.view).height-44, LSYSwiftDefine.ViewSize(self.view).width, 44)
    }
}